frontend/packages/metal3-plugin/src/components/host.tsx

```diff
@@ -208,9 +200,34 @@ export const BaremetalHostsPage: React.FC<BaremetalHostsPageProps> = (props) =>
       rowFilters={[hostStatusFilter]}
       createButtonText="Add Host"
       namespace={props.namespace}
-      resources={[hostsResource, machinesResource, nodesResource]}
+      resources={resources}
       flatten={flatten}
       ListComponent={HostList}
+      customData={{
+        hasNodeMaintenanceCapability: props.hasNodeMaintenanceCapability,
+      }}
     />
   );
 };
+
+const hostsPageStateToProps = ({ k8s }) => ({
+  hasNodeMaintenanceCapability: !!k8s.getIn([
+    'RESOURCES',
+    'models',
+    referenceForModel(NodeMaintenanceModel),
+  ]),
+});
+
+export const BaremetalHostsPageConnected = connect(hostsPageStateToProps)(BaremetalHostsPage);
+
+type BaremetalHostsPageProps = {
+  namespace: string;
+  hasNodeMaintenanceCapability: boolean;
+};
+
+type HostListProps = {
+  data: HostRowBundle[];
+  customData: {
```

Comment:
This object `{ hasNodeMaintenanceCapability: boolean; }` declaration is repeated on many places across many files.

Isn't it better to declare (in `types.ts`) and consistently reuse

```ts
type HostCustomData = {
  hasNodeMaintenanceCapability: boolean;
};
```

Reply:
again, IMO it is better to leave it for better extendability